db/src/main/java/com/readrops/db/queries/FoldersAndFeedsQueryBuilder.kt

```diff
@@ -51,11 +52,24 @@ object FoldersAndFeedsQueryBuilder {
                 }
             }""".trimIndent()
         )
+        return result
     }
 
-    private fun buildFeedQuery(accountId: Int, hideReadFeeds: Boolean): SupportSQLiteQuery {
-        val tables = if (hideReadFeeds) FEED_JOIN + ITEM_JOIN else FEED_JOIN
-        val selection = if (hideReadFeeds) FEED_SELECTION + ITEM_SELECTION else FEED_SELECTION
+    private fun buildFeedQuery(accountId: Int, hideReadFeeds: Boolean, useSeparateState: Boolean): SupportSQLiteQuery {
+        val tables = buildString {
+            append(FEED_JOIN)
+            if (hideReadFeeds) {
+                append(ITEM_JOIN)
+                if(useSeparateState) append(SEPARATE_STATE)
+            }
+        }
+        val selection = buildString {
+            append(FEED_SELECTION)
+            if (hideReadFeeds) {
+                if(useSeparateState) append("AND (ItemState.read = 0 OR ItemState.read IS NULL)")
```

Comment:
So I tested your solution and it didn't work for me, I still had some feeds and folders with 0 unread items displayed in the drawer.

And the catch is here. `ItemState` table isn't a perfect representation of the remote database. During synchronization only a limited read/unread ids are fetched. That's why when no state for an item is available in `ItemState`, we consider this item read. So in `AND (ItemState.read = 0 OR ItemState.read IS NULL)`, `ItemState.read IS NULL` shouldn't be here to get only feeds with unread items whose read state is available in `ItemState`.